hubblestack/extmods/modules/nebula_osquery.py

```diff
@@ -74,11 +74,52 @@ def queries(query_group,
         salt '*' nebula.queries hour verbose=True
         salt '*' nebula.queries hour pillar_key=sec_osqueries
     '''
+    config = []
+
+    if isinstance(query_file, list):
+        config['paths'] = query_file
+    else:
+        config['paths'] = [query_file]
+
+    config['verbose'] = verbose
+    global CONFIG_STALENESS
+    global CONFIG
+    if config.get('verbose'):
+        log.debug('Nebula beacon called.')
+        log.debug('Nebula beacon config from pillar:\n{0}'.format(config))
+    ret = []
+
+    # Get config(s) from salt fileserver if we don't have them already
+    if CONFIG and CONFIG_STALENESS < config.get('refresh_interval', 300):
+        CONFIG_STALENESS += 1
+        CONFIG.update(config)
+        CONFIG['verbose'] = config.get('verbose')
+        config = CONFIG
+    else:
+            if config.get('verbose'):
+            log.debug('No cached config found for nebula, retrieving fresh from fileserver.')
+        new_config = config
+        if isinstance(config.get('paths'), list):
+            for path in config['paths']:
+                if 'salt://' in path:
+                    path = __salt__['cp.cache_file'](path)
+                if os.path.isfile(path):
+                    with open(path, 'r') as f:
+                        query_file = _dict_update(new_config,
+                                                  yaml.safe_load(f),
+                                                  recursive_update=True,
+                                                  merge_lists=True)
+                else:
+                    log.error('Path {0} does not exist or is not a file'.format(path))
+        else:
+            log.error('Nebula beacon \'paths\' data improperly formatted. Should be list of paths')
+
```

Comment:
Looks like we have a bunch of extra pulsar stuff that we don't need here. We don't need to store the config in a global or worry about config staleness, because the most often nebula runs is every 15 minutes, so we can just read in the config fresh every time. Basically, all you need to do here is iterate over the list of config files, and use dictupdate to merge each of them into a starting empty dictionary, in order.

I recommend you use pulsar as a guide for when you get stuck, but I recommend you write this piece of code from scratch. It's different enough from pulsar to be worth keeping away from copy paste where possible. If you don't understand what needs to happen well enough to envision the logic, ping me and we can walk through it again.